src/v2i-hub/RSUHealthMonitorPlugin/src/RSUHealthMonitorWorker.h

```diff
@@ -0,0 +1,96 @@
+#pragma once
+#include <string>
+#include "RSU_MIB_4_1.h"
+#include <map>
+#include <vector>
+#include <memory>
+#include <algorithm>
+#include <nmeaparse/nmea.h>
+#include "PluginLog.h"
+#include <jsoncpp/json/json.h>
+#include "SNMPClient.h"
+#include <boost/algorithm/string/predicate.hpp>
+
+using namespace std;
+using namespace tmx::utils;
+using namespace tmx::utils::rsu41::mib::oid;
+
+namespace RSUHealthMonitor
+{
+    enum class RSUMibVersion
+    {
+        UNKOWN_MIB_V = 0,
+        RSUMIB_V_4_1 = 1,
+        RSUMIB_V_1218 = 2
+    };
+
+    struct RSUFieldOIDStruct
+    {
+        string field;
+        string oid;
+        bool required; // Indicate whether this field is required to before broadcasting the RSU status.
+    };
+
+    /**
+     * RSUStatusTable is custome defined RSU status information.
+     * The fields are the subset of fields from RSU MIB definition https://github.com/certificationoperatingcouncil/COC_TestSpecs/blob/master/AppNotes/RSU/RSU-MIB.txt
+     */
+    using RSUStatusConfigTable = vector<RSUFieldOIDStruct>;
+
+    class RSUHealthMonitorWorker
+    {
+    private:
+        // A map of RSU MIB version used and RSUStatusTable
+        shared_ptr<map<RSUMibVersion, RSUStatusConfigTable>> _RSUSTATUSConfigMapPtr;
```

Comment:
Why do we have a map between mib version and RSU Status Config table? Will we populate this map with all versions regardless of the configured version and then just use the configured version?

Reply:
yes, we will populate the map with all versions if exist like NTCIP1218 and RSU4.1. But the users can choose which version to use by specifying the mib version through the UI?